integration-tests/testconfig/testconfig.go

```diff
@@ -0,0 +1,360 @@
+package testconfig
+
+import (
+	"embed"
+	"encoding/base64"
+	"fmt"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker/test_env"
+	"os"
+	"strings"
+
+	"github.com/barkimedes/go-deepcopy"
+	"github.com/google/uuid"
+	"github.com/pelletier/go-toml/v2"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	ocr2_config "github.com/smartcontractkit/chainlink-starknet/integration-tests/testconfig/ocr2"
+	ctf_config "github.com/smartcontractkit/chainlink-testing-framework/config"
+	k8s_config "github.com/smartcontractkit/chainlink-testing-framework/k8s/config"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/osutil"
+)
+
+type GlobalTestConfig interface {
+	GetChainlinkImageConfig() *ctf_config.ChainlinkImageConfig
+	GetLoggingConfig() *ctf_config.LoggingConfig
+	GetNetworkConfig() *ctf_config.NetworkConfig
+}
+
+type UpgradeableChainlinkTestConfig interface {
+	GetChainlinkUpgradeImageConfig() *ctf_config.ChainlinkImageConfig
+}
+
+type CommonTestConfig interface {
+	GetCommonConfig() *Common
+}
+
+type Ocr2TestConfig interface {
+	GetOCR2Config() *ocr2_config.Config
+}
+
+type NamedConfiguration interface {
+	GetConfigurationName() string
+}
```

Comment:
curious: are you using any of these interfaces anywhere? if not, delete them. I added them to core for a very specific reason. @gheorghestrimtu was using some of the helper functions from `core`, but he wanted to use his own `TestConfig` struct with them. That's why I defined the interfaces in `core`, so that his `TestConfig` could implement them.

So even if you need to call functions from `core` you should implement interfaces defined there instead of declaring new ones.

Unless I'm missing something?

Reply:
Removed